geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/SimpleFeatureEncoder.scala

```diff
@@ -59,38 +59,85 @@ trait SimpleFeatureDecoder extends HasEncoding {
 
 object SimpleFeatureDecoder {
 
-  def apply(sft: SimpleFeatureType, encoding: FeatureEncoding) =
+  /**
+   * Decode without projecting.
+   *
+   * @param sft the encoded simple feature type to be decode
+   * @param encoding the encoding that was used to encode
+   * @param options any options that were used to encode
+   * @return a new [[SimpleFeatureDecoder]]
+   */
+  def apply(sft: SimpleFeatureType, encoding: FeatureEncoding, options: EncodingOption*) =
```

Comment:
I think you've essentially created 2 methods with the same signature. If you have two methods: a(s:String) a(s:String,i: Int = 0), it can't disambiguate a call to a("")